iceoryx_hoofs/include/iceoryx_hoofs/cxx/helplets.hpp

```diff
@@ -335,6 +335,12 @@ constexpr T from(const F value) noexcept;
 template <typename T, typename F>
 constexpr T into(const F value) noexcept;
 
+/// @brief Used to search for any string
```

Comment:
I would move this into `capro/service_description.hpp` since it is somehow part of that logic. The helplets have nothing to do with strings and string search.